src/libostree/ostree-repo-pull.c

```diff
@@ -4373,6 +4404,26 @@ ostree_repo_pull_with_options (OstreeRepo             *self,
         }
     }
 
+  if (opt_ref_keyring_map_set)
+    {
+      const gchar *collection_id, *ref_name, *keyring_remote_name;
+
+      while (g_variant_iter_loop (ref_keyring_map_iter, "(&s&s&s)", &collection_id, &ref_name, &keyring_remote_name))
+        {
+          g_autoptr(OstreeCollectionRef) c_r = NULL;
+
+          if (!ostree_validate_rev (ref_name, error))
```

Comment:
Also need to validate the `collection_id` (`ostree_validate_collection_id()`) and `keyring_remote_name` (`ostree_validate_remote_name()`).